src/card_searcher.rs

```diff
@@ -71,13 +73,35 @@ impl Search for &str {
 /// in a scryfall search parameters. The valid parameters can be seen
 /// [here](https://scryfall.com/docs/syntax).
 pub trait Param: fmt::Debug {
+    /// Adds a parameter's string version to the passed string
+    fn to_param(&self, f: &mut String);
+
     /// Turns a parameter into its string version.
-    fn to_param(&self) -> String;
-}
+    #[inline(always)]
+    fn to_param_string(&self) -> String {
+        let mut s = String::new();
+        self.to_param(&mut s);
+        s
+    }
+}
+
+assert_impl_all!(String: Param);
+assert_impl_all!(BooleanParam: Param);
+assert_impl_all!(StringParam: Param);
+assert_impl_all!(NumericParam: Param);
+assert_impl_all!(RarityParam: Param);
+assert_impl_all!(ColorParam: Param);
+assert_impl_all!(FormatParam: Param);
+assert_impl_all!(BorderColor: Param);
+assert_impl_all!(Frame: Param);
+assert_impl_all!(FrameEffect: Param);
+assert_impl_all!(GameParam: Param);
+assert_impl_all!(TimeParam: Param);
+assert_impl_all!(NotParam<String>: Param);
```

Comment:
What are these for?

Reply:
Meh, felt like it. We don't have tests, these aren't tests but at least it can catch some things